cvefeed/vuln.go

```diff
@@ -0,0 +1,125 @@
+package cvefeed
+
+import (
+	"github.com/facebookincubator/nvdtools/wfn"
+)
+
+// Vuln is a vulnerability interface
+type Vuln interface {
+	// vulnerability should also be able to match attributes
+	wfn.Matcher
+	// ID returns the vulnerability ID
+	ID() string
+	// CVEs returns all CVEs it includes/references
+	CVEs() []string
+	// CWEs returns all CWEs for this vulnerability
+	CWEs() []string
+	// CVSSv2BaseScore returns CVSS v2 base score
+	CVSSv2BaseScore() float64
+	// CVSSv2BaseScore returns CVSS v2 vector
+	CVSSv2Vector() string
+	// CVSSv2BaseScore returns CVSS v3 base score
+	CVSSv3BaseScore() float64
+	// CVSSv2BaseScore returns CVSS v3 vector
+	CVSSv3Vector() string
+}
+
+// MergeVuln combines two Vulns:
+// resulted Vuln inherits all mutually exclusive methods (e.g. ID()) from Vuln x;
+// functions returning CVEs and CWEs return distinct(union(x,y))
+// the returned vuln matches attributes if x matches AND y doesn't
+func MergeVulns(x, y Vuln) Vuln {
```

Comment:
@skogtwin I'd like to change this before I merge
```
type overriden struct {
  Vuln
  m wfn.Matcher
}

// Match() and Config() would be overriden from m, because of clashing names

// instead of merge
fun OverrideVuln(v, override Vuln) Vuln {
  return &overriden{v, &andMatcher{v, wfn.DontMatch(override)}}
}
```

And that's it. I mean, it can inherit everything from the original one, or at least I think it should.
The only change should be how the matching is done, but score, vector, cves[] ... should be just from the original one.
yes or no? can leave it like this if you think it's worth it.

And make it return struct overridenVuln{v,

Reply:
IMHO, cosmetics. If you like this way better, sure, go for it.